### ISSUE TITLE
switch-default shouldn't suggest an unreachable default block

### ISSUE THREAD

**Issue by danvk** [palantir/tslint#2104]
### Bug Report

- __TSLint version__: `4.3.1`
- __TypeScript version__: `2.1.5`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
type Tag = 'foo' | 'bar';
function processTag(tag: Tag) {
  switch (tag) {
    case 'foo': return 1;
    case 'bar': return 2;
  }
}
```

with `tslint.json` configuration:

```json
{
  "switch-default": true
}
```

#### Actual behavior

tslint flags the switch statement: `Switch statement should include a 'default' case`.

#### Expected behavior

The switch/case is exhaustive. A default branch is unnecessary.

Related to #696 but with type unions instead of enums.

**Comment by burtek** [comment]
Duplicate of #696 ?

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
